### PR TITLE
Add reading specific annotation files for train dataset in Cityscapes

### PR DESCRIPTION
**Pull request by yasakova-anastasia**
### Summary

- Added special labelmap in Cityscapes for training dataset
- `background` can be at any index instead of just `0` in Cityscapes, VOC

### How to test

Unit test and manually on real dataset

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md)
- [ ] I have updated the [documentation](
  https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly
- [x] I have added tests to cover my changes
- [ ] I have [linked related issues](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Review by zhiltsov-max — CHANGES_REQUESTED** [review]

File: datumaro/plugins/cityscapes_format.py [hunk 1]
```diff
@@ -71,24 +95,26 @@ class CityscapesPath:
     GT_INSTANCE_MASK_SUFFIX = '_' + GT_FINE_DIR + INSTANCES_IMAGE
     COLOR_IMAGE = '_color.png'
     LABELIDS_IMAGE = '_labelIds.png'
+    LABEl_TRAIN_IDS_SUFFIX = '_' + GT_FINE_DIR + '_labelTrainIds.png'
 
     LABELMAP_FILE = 'label_colors.txt'
 
-def make_cityscapes_categories(label_map=None):
+def make_cityscapes_categories(label_map=None, use_train_label_map=False):
```
[the remark this answers is not in the document]
Comment (yasakova-anastasia) [reply datumaro/plugins/cityscapes_format.py]: Removed the second parameter.

File: datumaro/plugins/cityscapes_format.py [hunk 2]
```diff
@@ -170,19 +196,17 @@ def __init__(self, path, subset=None):
 
         super().__init__(subset=subset)
 
-        self._categories = self._load_categories(self._path)
         self._items = list(self._load_items().values())
 
-    def _load_categories(self, path):
-        label_map = None
+    def _load_categories(self, path, label_map=None):
```
[the remark this answers is not in the document]
Comment (yasakova-anastasia) [reply datumaro/plugins/cityscapes_format.py]: Removed `label_map` parameter and added `use_train_label_map`  parameter.

File: datumaro/plugins/cityscapes_format.py [hunk 3]
```diff
@@ -170,19 +192,21 @@ def __init__(self, path, subset=None):
...
             label_map_path = osp.join(path, CityscapesPath.LABELMAP_FILE)
             if osp.isfile(label_map_path):
                 label_map = parse_label_map(label_map_path)
-            else:
-                label_map = CityscapesLabelMap
+            elif label_map is None:
+                if use_train_label_map:
+                    label_map = TRAIN_CITYSCAPES_LABEL_MAP
+                else:
+                    label_map = CITYSCAPES_LABEL_MAP
```
Comment (zhiltsov-max) [inline datumaro/plugins/cityscapes_format.py:209]:
```suggestion

        if label_map is None:
            if use_train_label_map:
                label_map = TRAIN_CITYSCAPES_LABEL_MAP
            else:
                label_map = CITYSCAPES_LABEL_MAP
```

File: datumaro/plugins/cityscapes_format.py [hunk 4]
```diff
@@ -237,6 +264,12 @@ def _load_items(self):
             items[item_id] = DatasetItem(id=item_id, subset=self._subset,
                 image=path)
 
+        if mask_suffix is CityscapesPath.LABEL_TRAIN_IDS_SUFFIX:
+            self._categories = self._load_categories(self._path,
+                use_train_label_map=True)
+        else:
+            self._categories = self._load_categories(self._path)
+
```
Comment (zhiltsov-max) [inline datumaro/plugins/cityscapes_format.py:272]:
```suggestion
        self._categories = self._load_categories(self._path,
            use_train_label_map= \
                mask_suffix is CityscapesPath.LABEL_TRAIN_IDS_SUFFIX)
```

File: tests/test_cityscapes_format.py [hunk 5]
```diff
@@ -109,6 +111,55 @@ def test_can_import(self):
...
+            ),
+            DatasetItem(id='defaultcity/defaultcity_000001_000019',
+                subset = 'val',
+                image=np.ones((1, 5, 3)),
+                annotations=[
+                    Mask(np.array([[1, 1, 1, 1, 1]]), label=19,
+                        attributes={'is_crowd': True}),
+                ]
+            ),
+        ], categories=Cityscapes.make_cityscapes_categories(label_map=TRAIN_CITYSCAPES_LABEL_MAP))
```
Comment (zhiltsov-max) [inline tests/test_cityscapes_format.py:157]:
```suggestion
        ], categories=Cityscapes.make_cityscapes_categories(
            label_map=TRAIN_CITYSCAPES_LABEL_MAP))
```

**Review by zhiltsov-max — APPROVED** [review]

**Pull request merged**